### ISSUE TITLE
Randomly Devices are Added to Homeykit

### ISSUE THREAD

**Issue by BBEHannes** [swttt/com.swttt.homekit#82]
I have been seeing this lately where in Settings, devices that i specifically unchecked from showing up in Homekit are added again after a while. This is ok as long as there is no restart of Homey or the App since these changes are not enforced until restart but afterwards it costs a lot of time (since i have a lot of devices) to get them set properly again.

**Comment by robertklep** [comment]
Strange, I have	about 25 devices in total that Homeykit can manage, and I've never had an issue with devices suddenly changing their "exposed" status.

It's especially strange that the device state as shown on the settings page doesn't reflect the actual status of the device (because, like you say, the device is being shown as exposed, but until a reboot/restart it isn't treated as such).

For now, I don't have a clue how this might happen.

Do you use the Homey desktop app, or a browser? If so, it might be worthwhile to open the developer tools to check if there are any errors happening when you change the state of a device. Perhaps the frontend is having issues syncing the device status with the backend, for some reason.

**Comment by BBEHannes** [comment]
I agree with everything Robert :). I use the homey desktop App. Haven't played around with the developer tools yet as of now. It is very inconsistent behavior and does not occur all the time. is there anything i can do to help you provide logs at times of the behavior? if so, please let me know

**Comment by robertklep** [comment]
Too bad the behaviour is seemingly random, that means the devtools are of limited use I think.

Have you tried changing the state of a device on the settings page when it has reverted back to being exposed (while it shouldn't)?

An additional complication is that the device state, whether it should be shown or not, is not handled by the Homey app directly but by the underlying `hap-nodejs` module.

@nhassink @swttt have you ever had to deal with this problem?

**Comment by BBEHannes** [comment]
It MIGHT change when adding new devices to Homey. I am thinking about the conditions and I believe that it happens when I have added new physical devices. Cannot confirm it yet though. 

Indeed, what I do when this occurs is set the state back to what it is supposed to be.

**Comment by nhassink** [comment]
If you add a new physical device to homey it will automatically appear in HomeKit / home app. Might this be the behavior you’re experiencing?

**Comment by BBEHannes** [comment]
Thanks hassink. It is the exact behavior I mentioned in my initial comment and verified by Robert. All devices show as “DELETE” under their name. This is wrong info as most of them should not be added to HomeKit view. And they are not, until you restart the app or homey.

**Comment by robertklep** [comment]
I might have found a possible reason why this happens, but only if there's something wrong with some core Homey code.

However, I'll try and cut a beta release tomorrow or on monday to see if that solves your problem.

**Comment by BBEHannes** [comment]
Rockstar! Looking forward seeing the results

**Comment by BBEHannes** [comment]
Friendly reminder on this.

**Comment by BBEHannes** [comment]
This is still somewhat of an issue. When restarting homey without first shutting down HomeKit, some of my devices are lost and I need to re-add them to homeykit after restart.

**Comment by robertklep** [comment]
This should be fixed in an upcoming release (probably v3.0.7)

**Comment by FrankRinkens** [comment]
Don't see this happening in v3.07 anymore, @BBEHannes pls check this beta. ;)

**Comment by BBEHannes** [comment]
Still running 3.06 but it happens there (intermittently, not 100% of the time) still.

**Comment by robertklep** [comment]
@BBEHannes it was (hopefully) fixed in 3.0.7, not 3.0.6.

**Comment by BBEHannes** [comment]
Updated to the beta... will keep you posted.
